x-pack/plugins/security_solution/public/detections/pages/detection_engine/rules/all/rules_table_utility_bar.tsx

```diff
@@ -125,91 +123,69 @@ export const AllRulesUtilityBar = React.memo<AllRulesUtilityBarProps>(
       <UtilityBar border>
         <UtilityBarSection>
           <UtilityBarGroup>
-            {hasBulkActions ? (
-              <UtilityBarText dataTestSubj="showingRules">
-                {i18n.SHOWING_RULES(paginationTotal)}
-              </UtilityBarText>
-            ) : (
-              <UtilityBarText dataTestSubj="showingExceptionLists">
-                {i18n.SHOWING_EXCEPTION_LISTS(paginationTotal)}
-              </UtilityBarText>
-            )}
+            <UtilityBarText dataTestSubj="showingRules">
+              {i18n.SHOWING_RULES(...getShowingRulesParams(pagination))}
+            </UtilityBarText>
           </UtilityBarGroup>
+          <>
+            <UtilityBarGroup data-test-subj="tableBulkActions">
+              <UtilityBarText dataTestSubj="selectedRules">
+                {i18n.SELECTED_RULES(numberSelectedItems)}
+              </UtilityBarText>
 
-          {hasBulkActions ? (
-            <>
-              <UtilityBarGroup data-test-subj="tableBulkActions">
-                <UtilityBarText dataTestSubj="selectedRules">
-                  {i18n.SELECTED_RULES(numberSelectedItems)}
-                </UtilityBarText>
-
-                {canBulkEdit && onToggleSelectAll && hasPagination && (
-                  <UtilityBarAction
-                    disabled={hasDisabledActions}
-                    dataTestSubj="selectAllRules"
-                    iconType={isAllSelected ? 'cross' : 'pagesSelect'}
-                    iconSide="left"
-                    onClick={onToggleSelectAll}
-                  >
-                    {isAllSelected ? i18n.CLEAR_SELECTION : i18n.SELECT_ALL_RULES(paginationTotal)}
-                  </UtilityBarAction>
-                )}
-
-                {canBulkEdit && (
-                  <UtilityBarAction
-                    disabled={hasDisabledActions}
-                    inProgress={isBulkActionInProgress}
-                    dataTestSubj="bulkActions"
-                    iconSide="right"
-                    iconType="arrowDown"
-                    popoverPanelPaddingSize="none"
-                    popoverContent={handleGetBulkItemsPopoverContent}
-                  >
-                    {i18n.BATCH_ACTIONS}
-                  </UtilityBarAction>
-                )}
-
+              {canBulkEdit && onToggleSelectAll && (
```

Comment:
I believe after splitting the utility bar into two components, we could make the `onRefresh`, `onRefreshSwitch`, and `onToggleSelectAll` props required. And remove all guards associated with them.